### ISSUE TITLE
For Radio Button Lists with model as object, on show after hide, the selection is cleared, but the model array still holds the value

### ISSUE THREAD

**Issue by cibinthomasjacob** [ngx-formly/ngx-formly#2381]
**For Radio Button Lists with model as object, on show after hide, the selection is cleared, but the model array still holds the value**

We are building dynamic JSON based forms , have a use case for Radio button with valueProp: (option) => option to hold model as object . Ran into an issue where the radiobutton will not retain its selection if displayed back after getting hidden using hideexpression. The model array will still have the selected item from when it was displayed. This is breaking other expressions which refers to the model value while its hidden. How to clear the value on hide, or how to repopulate the selected item from model on show? The actual use case is more complex than provided in the example .
 
Example: https://stackblitz.com/edit/ngx-formly-ui-material-ya7cxk

**Comment by aitboudad** [comment]
Formly doesn't auto-clear the hidden fields, I'm going to introduce an option in order to enable that behavior which I think it should be enabled by default in the future major version.

For a workaround pass `autoClear` to field config:

```patch
    {
      key: 'Radio1',
      type: 'radio',
      hideExpression:'model.radio?.value != "1"',
      templateOptions: { ... },
-   }
+     autoClear: true,
+   } as any,
```

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.0](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.0) release. You can now enable clearing the hidden field value using the following config:

```patch
    FormlyModule.forRoot({
+      extras: {
+        resetFieldOnHide: true,
+      },
    }),
```

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
